tests/acceptance/standard/test_job_token_scope.py

```diff
@@ -31,25 +31,60 @@ def test__enable_limit_access_to_this_project(
 
     def test__disable_limit_access_to_this_project(
         self,
+        gl,
         project: Project,
     ):
-        self._setup_limit_access_state(project, True)
-
-        job_token_scope = f"""
-        projects_and_groups:
-          {project.path_with_namespace}:
-            job_token_scope:
-              limit_access_to_this_project: false
         """
+        Test that project-level 'Limit access to this project' setting can be configured
+        when the instance-level setting 'enforce_ci_inbound_job_token_scope_enabled' is disabled.
 
-        run_gitlabform(job_token_scope, project)
+        The test verifies that:
+        1. When instance setting is disabled, projects can control their own 'inbound_enabled' setting
+        2. The gitlabform config 'limit_access_to_this_project: false' correctly sets
+           the project's 'inbound_enabled' to false via the GitLab API
 
-        scope = project.job_token_scope.get()
-
-        # inbound_enabled is what GL returns from GET
-        # https://docs.gitlab.com/ee/api/project_job_token_scopes.html#get-a-projects-cicd-job-token-access-settings
-        # to denote whether "Limit Access to this Project" is enabled or not
-        assert not scope.inbound_enabled
+        This test requires admin access to modify the instance setting.
+        """
+        # Setup: Disable instance-level enforcement to allow project-level configuration
+        instance_settings = gl.settings.get()
+        instance_settings.enforce_ci_inbound_job_token_scope_enabled = False
+
+        # Retry settings update in case of temporary resource issues
+        max_retries = 3
+        for attempt in range(max_retries):
+            try:
+                instance_settings.save()
+                break
+            except Exception as e:
+                if attempt == max_retries - 1:
+                    raise Exception(f"Failed to update instance settings after {max_retries} attempts: {str(e)}")
+                print(f"Attempt {attempt + 1} failed, retrying...")
```

Comment:
Do we really need this retry loop even though [we already use `retry_transient_errors`](https://github.com/gitlabform/gitlabform/issues/1025#issuecomment-2894233400)?

Reply:
Good question. Not sure if we really need this or not.

When I was running that test locally (not the entire suite), I'd occasionally get '500 Internal server error' when attempting to update gitlab application/instance setting.

This was probably because gitlab running in docker, but not sure. Since we do the same in the CI process, thought of adding the retry so that test is not randomly failing making the pipeline unstable.

Let me know if you have suggestions.

Reply:
`retry_transient_errors` should help with 500s, quoting [the docs](https://python-gitlab.readthedocs.io/en/stable/api-usage-advanced.html#transient-errors):

> GitLab server can sometimes return a transient HTTP error. python-gitlab can automatically retry in such case, when retry_transient_errors argument is set to True. When enabled, HTTP error codes 500 (Internal Server Error), 502 (502 Bad Gateway), 503 (Service Unavailable), 504 (Gateway Timeout), and Cloudflare errors (520-530) are retried.

Reply:
Yup. I agree with you. But, I'm just not sure if that is sufficient for gitlabform tests which uses gitlab in a container with very limited resources compared to how gitlab would be running in a real environment.

I'm happy to remove the retry logic and we can re-evaluate it if it doesn't work consistently in gitlabform ci, running gitlab in container.

Reply:
@gdubicki - Going to resolve this thread and merge this PR so that other PRs are not blocked because of broken CI pipeline. If you find something wasn't addressed here, please open a new issue.

Thanks for your review.